Northstar.Client/mod/scripts/vscripts/ui/menu_mod_settings.nut

```diff
@@ -0,0 +1,1105 @@
+		child.SetNavDown( Hud_GetChild( file.modPanels[ int( PureModulo( i + 1, len ) ) ], "Slider" ) )
+
+		file.sliders.append( MS_Slider_Setup( child ) )
+
+		Hud_AddEventHandler( child, UIE_CHANGE, OnSliderChange )
+
+		child = Hud_GetChild( panel, "OpenCustomMenu" )
+
+		Hud_AddEventHandler( child, UIE_CLICK, CustomButtonPressed )
+	}
+
+	// Hud_AddEventHandler( Hud_GetChild( file.menu, "BtnModsSearch" ), UIE_LOSE_FOCUS, OnFilterTextPanelChanged )
+	Hud_AddEventHandler( Hud_GetChild( file.menu, "BtnFiltersClear" ), UIE_CLICK, OnClearButtonPressed )
+	// mouse delta
+	AddMouseMovementCaptureHandler( file.menu, UpdateMouseDeltaBuffer )
+
+	Hud_AddEventHandler( Hud_GetChild( file.menu, "BtnModsSearch" ), UIE_CHANGE, void function ( var inputField ) : ()
+	{
+		file.filterText = Hud_GetUTF8Text( inputField )
+		OnFiltersChange()
+	} )
+}
+
+// "PureModulo"
+// Used instead of modulo in some places.
+// Why? beacuse PureModulo loops back onto positive numbers instead of going into the negatives.
+// DO NOT TOUCH.
+// a / b != floor( float( a ) / b )
+// int( float( a ) / b ) != floor( float( a ) / b )
+// Examples:
+// -1 % 5 = -1
+// PureModulo( -1, 5 ) = 4
+float function PureModulo( int a, int b )
+{
+	return b * ( ( float( a ) / b ) - floor( float( a ) / b ) )
+}
+
+void function ResetConVar( var button )
+{
+	ConVarData conVar = file.filteredList[ int ( Hud_GetScriptID( Hud_GetParent( button ) ) ) + file.scrollOffset ]
+
+	if ( conVar.isCategoryName )
+		ShowAreYouSureDialog( "#ARE_YOU_SURE", ResetAllConVarsForModEventHandler( conVar.catName ), "#WILL_RESET_ALL_SETTINGS"  )
+	else
+		ShowAreYouSureDialog( "#ARE_YOU_SURE", ResetConVarEventHandler( int ( Hud_GetScriptID( Hud_GetParent( button ) ) ) + file.scrollOffset ), Localize( "#WILL_RESET_SETTING", Localize( conVar.displayName ) ) )
+}
+
+void function ShowAreYouSureDialog( string header, void functionref() func, string details )
+{
+	DialogData dialogData
+	dialogData.header = header
+	dialogData.message = details
+
+	AddDialogButton( dialogData, "#NO" )
+	AddDialogButton( dialogData, "#YES", func )
+
+	AddDialogFooter( dialogData, "#A_BUTTON_SELECT" )
+	AddDialogFooter( dialogData, "#B_BUTTON_BACK" )
+
+	OpenDialog( dialogData )
+}
+
+void functionref() function ResetAllConVarsForModEventHandler( string catName )
+{
+	return void function() : ( catName )
+	{
+		for ( int i = 0; i < file.conVarList.len(); i++ )
+		{
+			ConVarData c = file.conVarList[i]
+			if ( c.catName != catName || c.isCategoryName || c.isEmptySpace || c.isCustomButton )
+				continue
+			SetConVarToDefault( c.conVar )
+
+			int index = file.filteredList.find( c )
+			if ( file.filteredList.find( c ) < 0 )
+				continue
+
+			if ( min( BUTTONS_PER_PAGE, max( 0, index - file.scrollOffset ) ) == index - file.scrollOffset )
+			{
+				Hud_SetText( Hud_GetChild( file.modPanels[ i - file.scrollOffset ], "TextEntrySetting" ), c.isEnumSetting ? c.values[ GetConVarInt( c.conVar ) ] : GetConVarString( c.conVar ) )
```

Comment:
Sometimes when resetting a category it throws an error (it's the published version on thunderstore so the line number is a bit different but I see no difference to this pr):
```
[SCRIPT UI] [info] SCRIPT ERROR: [UI] Index "15" is beyond array size of 15
[SCRIPT UI] [info]  -> Hud_SetText( Hud_GetChild( file.modPanels[ i - file.scrollOffset ], "TextEntrySetting" ), c.isEnumSetting ? c.values[ GetConVarInt( c.conVar ) ] : GetConVarString( c.conVar ) )
[SCRIPT UI] [info] 
CALLSTACK
*FUNCTION [unknown()] ui/menu_mod_settings.nut line [237]
*FUNCTION [call()] NATIVE line [-1]
*FUNCTION [OnDialogButton_Activate()] ui/menu_dialog.nut line [134]
```

`BUTTONS_PER_PAGE` is 15 so `file.modPanels[BUTTONS_PER_PAGE]` is out of bounds. I don't have access to this pr so you can introduce my fix [here](https://github.com/Glacir/ModSettings/commit/b451f958293f8a91f3f84eda285596fdd0fc2616).

Reply:
Can you elaborate when this happens? I'll push it regardless but I don't see when this could overflow

Reply:
resolved in a46f31c

Reply:
`i - file.scrollOffset` line 247 look like it should be `index - file.scrollOffset `. I don't have a chance to test now but plz take a look.

Reply:
Good catch. I was able to cause a script error because of this.

Reply:
resolved in 13202a6